ArtOfIllusion/src/artofillusion/LayoutWindow.java

```diff
@@ -439,59 +450,58 @@ else if ("-".equals(lines[i]))
 
   private void createFileMenu()
   {
-    BMenuItem item;
     BMenu importMenu, exportMenu;
-    List<Translator> trans = PluginRegistry.getPlugins(Translator.class);
+    List<Translator> translators = PluginRegistry.getPlugins(Translator.class);
 
     fileMenu = Translate.menu("file");
-    menubar.add(fileMenu);
+    
     importMenu = Translate.menu("import");
     exportMenu = Translate.menu("export");
-    fileMenuItem = new BMenuItem [1];
-    fileMenu.add(Translate.menuItem("new", this, "actionPerformed"));
-    fileMenu.add(Translate.menuItem("open", this, "actionPerformed"));
-    fileMenu.add(recentFilesMenu = Translate.menu("openRecent"));
-    RecentFiles.createMenu(recentFilesMenu);
-    fileMenu.add(Translate.menuItem("close", this, "actionPerformed"));
+    
+    fileMenu.add(Translate.menuItem("new", this, "newSceneAction"));
+    fileMenu.add(Translate.menuItem("open", this, "openSceneAction"));
+    fileMenu.add(recentFilesMenu = RecentFiles.createRecentMenu());
+    
+    fileMenu.add(Translate.menuItem("close", this, "closeSceneAction"));
     fileMenu.addSeparator();
-    Collections.sort(trans, new Comparator<Translator>() {
-      @Override
-      public int compare(Translator o1, Translator o2)
-      {
-        return o1.getName().compareTo(o2.getName());
-      }
-    });
-    for (int i = 0; i < trans.size(); i++)
-      {
-        if (trans.get(i).canImport())
-          {
-            importMenu.add(item = new BMenuItem(trans.get(i).getName()));
-            item.setActionCommand("import");
-            item.addEventLink(CommandEvent.class, this, "actionPerformed");
-          }
-        if (trans.get(i).canExport())
-          {
-            exportMenu.add(item = new BMenuItem(trans.get(i).getName()));
-            item.setActionCommand("export");
-            item.addEventLink(CommandEvent.class, this, "actionPerformed");
-          }
-      }
+    
+    Collections.sort(translators, Comparator.comparing(Translator::getName));
+    for (Translator translator: translators)
+    {
+        if(translator.canImport())
+        {
+          BMenuItem tim = new BMenuItem(translator.getName());
```

Comment:
Who's tim? :)

Seriously, is there a reason that you used this name for the variable?